src/intervals/interval_operations/numeric.jl

```diff
@@ -88,7 +88,7 @@ See also: [`inf`](@ref), [`sup`](@ref), [`bounds`](@ref), [`diam`](@ref),
 [`radius`](@ref) and [`midradius`](@ref).
 """
 function mid(x::BareInterval{T}, α = 0.5) where {T<:AbstractFloat}
-    0 ≤ α ≤ 1 || throw(DomainError(α, "α must be between 0 and 1"))
+    0 ≤ α ≤ 1 || return throw(DomainError(α, "α must be between 0 and 1"))
```

Comment:
Do we need the return here? I just wonder if that would cause some sort of type unstability (sometimes, the returned type would be an `Interval`, or a `DomainError`)

(Similar comment applies below.)

Reply:
I think it is safe. I checked `@code_warntype` for

```Julia
function foo(x)
    x < 1 && return throw(DomainError("x is smaller than 1"))
    return true
end

function goo(x)
    x < 1 && throw(DomainError("x is smaller than 1"))
    return true
end
```

and the two functions seem to be compiled equivalently.